### PR TITLE
refactor: replace next id storage provider with storage service

### PR DESCRIPTION
**Pull request by nuanyang233**
- fix: add signer parameter in NextID storage
- fix: common logic

## Description

Closes #MF-1852 MF-1559

## Type of change

- [ ] Documentation
- [x] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/plugins/Game/hook/useGameList.ts [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
+import { useWeb3State } from '@masknet/plugin-infra/web3'
 import { useGameConstants } from '@masknet/web3-shared-evm'
 import { EMPTY_LIST } from '@masknet/shared-base'
-import { PluginGameRPC } from '../messages'
-import type { GameInfo } from '../types'
+import type { GameRSSNode } from '../types'
 
 export function useGameList() {
-    const connection = useWeb3Connection(NetworkPluginID.PLUGIN_EVM)
+    const { Storage } = useWeb3State()
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Game/hook/useGameList.ts]: If the currently selected network isn't EVM, then `Storage.createRSSStorage` will throw an error. Cause the RSS3 storage only implemented across EVM chains.

File: packages/mask/src/plugins/Pets/hooks/useUser.ts [hunk 2]
```diff
@@ -20,12 +20,16 @@ export function useUser() {
 export function useCurrentVisitingUser(flag?: number) {
     const [user, setUser] = useState<User>({ userId: '', address: '' })
     const identity = useCurrentVisitingIdentity()
+    const { Storage } = useWeb3State()
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Pets/hooks/useUser.ts:23]: Does it necessary?

```suggestion
    const { Storage } = useWeb3State(NetworkPluginID.PluginID_EVM)
```

Comment (guanbinrui) [reply packages/mask/src/plugins/Pets/hooks/useUser.ts:23]: Please also add `Storage` to the dependencies list.

File: packages/plugins/Web3Profile/src/SNSAdaptor/types.ts [hunk 3]
```diff
@@ -30,32 +29,32 @@ export interface PersonaKV {
...
+//     hiddenAddresses: WalletsCollection
+//     unListedCollections: Record<
+//         string,
+//         {
+//             NFTs: string[]
+//             Donations: string[]
+//             Footprints: string[]
+//         }
+//     >
+// }
```
Comment (guanbinrui) [inline packages/plugins/Web3Profile/src/SNSAdaptor/types.ts]:
```suggestion
```

File: packages/plugins/Web3Profile/src/SNSAdaptor/types.ts [hunk 4]
```diff
@@ -30,32 +29,32 @@ export interface PersonaKV {
...
+// export interface CollectionTypes {
+//     platform: NetworkPluginID
+//     address: string // take id as address if collection is a poap
+//     key: string // address + tokenId as unique key of NFT, id as unique key of poap
+//     tokenId?: string
+//     iconURL?: string
+//     hidden?: boolean
+//     name?: string
+//     chainId?: ChainId
+// }
```
Comment (guanbinrui) [inline packages/plugins/Web3Profile/src/SNSAdaptor/types.ts]:
```suggestion
```

File: packages/shared/src/hooks/useWeb3ProfileHiddenSetting.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import { useWeb3State } from '@masknet/plugin-infra/web3'
+import { PluginId } from '@masknet/plugin-infra'
+import { NextIDPlatform, EMPTY_LIST } from '@masknet/shared-base'
+import { isSameAddress } from '@masknet/web3-shared-base'
+import { useAsync } from 'react-use'
+import { useMemo } from 'react'
+import type { CollectionType } from '@masknet/web3-providers'
+import type { Web3ProfileStorage } from '../types'
+
+export function useWeb3ProfileHiddenSetting(
```
Comment (guanbinrui) [inline packages/shared/src/hooks/useWeb3ProfileHiddenSetting.ts]:
```suggestion
export function useWeb3ProfileHiddenSettings(
```

File: packages/shared/src/hooks/useWeb3ProfileHiddenSetting.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    identity?: string,
+    publicKeyAsHex?: string,
+    options?: {
+        address?: string
+        hiddenAddressesKey?: 'NFTs' | 'donations' | 'footprints'
+        collectionKey?: CollectionType
+    },
+) {
+    const { Storage } = useWeb3State()
```
Comment (guanbinrui) [inline packages/shared/src/hooks/useWeb3ProfileHiddenSetting.ts]:
```suggestion
    const { Storage } = useWeb3State(NetworkPluginID.PLUGIN_EVM)
```

File: packages/mask/src/plugins/Pets/hooks/useUser.ts (hunk 2 above)
Comment (nuanyang233) [reply packages/mask/src/plugins/Pets/hooks/useUser.ts:23]: I think it isn't necessary. KV Storage is support on all network.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**